dali/bus.py

```diff
@@ -130,11 +128,93 @@ def assign_short_addresses(self):
                     i.send(gear.ProgramShortAddress(new_addr))
                     r = i.send(gear.VerifyShortAddress(new_addr))
                     if r.value is not True:
-                        raise ProgramShortAddressFailure(new_addr)
+                        ProgramShortAddressFailure(new_addr)
+                        print(f"Error in programming short address {new_addr}")
```

Comment:
`print` is probably some leftover debug code ?

Reply:
original code used raise, which crashed the software. I replaced the raise with printf. It is not acceptable that the code crash if verifying short address fails

Reply:
The idea is that the caller must be aware of the possibility of programming the short address failing, and must catch the exception if they wish to continue in that circumstance.

Reply:
We should check why it crashes, `print` inside a daemon software is not good option. 

Missing a `try ... except` around the call to `search_bus` ?

Reply:
I will try to find a way to handle the exception in QT code

Reply:
bus.py reverted back to the original one. required functionality implemented in hasseb.py